src/main/java/ch/njol/skript/effects/EffBan.java

```diff
@@ -81,7 +82,7 @@ public boolean init(final Expression<?>[] exprs, final int matchedPattern, final
 	@Override
 	protected void execute(final Event e) {
 		final String reason = this.reason != null ? this.reason.getSingle(e) : null; // don't check for null, just ignore an invalid reason
-		final Date expires = null;
+		final Date expires = duration; // pseudo code

```

Comment:
``duration`` is a ``Timespan`` object, while ``expires`` is a ``Date`` object. Instead, use:
```
final Date expires = new Date(System.currentTimeMillis() + duration.getMilliSeconds());
```
*I think this is the right way to create the Date object, haven't worked with Date objects before... if it's wrong someone please correct me lol*